module/zfs/zio_inject.c

```diff
@@ -699,6 +729,10 @@ zio_handle_io_delay(zio_t *zio)
 		 */
 		min_handler->zi_next_lane = (min_handler->zi_next_lane + 1) %
 		    min_handler->zi_record.zi_nlanes;
+
+		min_handler->zi_record.zi_match_count++;
```

Comment:
I think match could be counted in the loop above, so that we could count several matches before selecting only one inject.  It just needs `freq_triggered()` to be moved some lower after other checks.  And once match counter become useful, `print_delay_handler()` could print it also, also unifying it with other types.

Reply:
Oh yeah, neat! Trying now!

Reply:
Yeah, super nice!

```
 ID  POOL             GUID              DELAY (ms)  LANES  FREQ       MATCH   INJECT
---  ---------------  ----------------  ----------  -----  ---------  ------  ------
  1  tank             6889df1eaa5c864a          50      1  100.0000%      79      39
  2  tank             6889df1eaa5c864a         100      2  100.0000%      79      18
  3  tank             6889df1eaa5c864a         150      4  100.0000%      79      22
  4  tank             6889df1eaa5c864a         200      8  100.0000%      79       0
```

New commit added, top post updated. Great suggestion, thanks!